### ISSUE TITLE
bug: events are not showing up in any dropdown or searches

### ISSUE THREAD

**Issue by tanmoysrt** [PostHog/posthog#26531]
### Bug Description

We have a self-hosted posthog instance (PostHog Git SHA - 5adf41e8b8).
In activity pages, we can see the incoming events and event types as well.

But while creating insights in dashboard, the dropdown to select event is blank.
It's happenning on the activity page's `Select an event` dropdown as well. 

## How to reproduce

1. Go to dashboard
2. Try to create an insight
3. In the event selection dropdown it's showing up no events

   ![Screenshot 2024-11-29 at 10 49 35 AM](https://github.com/user-attachments/assets/ca838c0c-06bd-4440-a477-dc4647beea15)

## Additional context
1. In our posthog instance, there are already various events, checked in data warehouse and in activity page (/project/1/activity/explore)

   ![Screenshot 2024-11-29 at 10 51 30 AM](https://github.com/user-attachments/assets/610f5897-6336-48bb-a3de-b2c8beaf6f2e)

3. We have checked the API call as well, it returning blank results
    **Request** - 
    ```
    https://<posthog_instance>/api/projects/1/event_definitions?event_type=event&search=&limit=100&offset=0
    ```

   **Response** -
    ```json
   {
     "count": 0,
     "next": null,
     "previous": null,
     "results": []
   }
   ```

### Debug info

- [ ] PostHog Cloud, Debug information: [please copy/paste from https://us.posthog.com/settings/project-details#variables or https://eu.posthog.com/settings/project-details#variables]
- [x] PostHog Hobby self-hosted with `docker compose`, version/commit: 5adf41e8b8
- [ ] PostHog self-hosted with Kubernetes (deprecated, see [`Sunsetting Kubernetes support`](https://posthog.com/blog/sunsetting-helm-support-posthog)), version/commit: [please provide]

**Comment by kenz16** [comment]
I face the same problem. and still struggling how to solve the issue.. please help us posthog team
Thanks in advance

<img width="884" alt="Screenshot 2024-11-29 at 19 22 20" src="https://github.com/user-attachments/assets/2ee52268-04a5-4258-8741-8f6fd8b5075c">

**Comment by DavidKuennen** [comment]
Having the same issue.

**Comment by SanchoKrancho** [comment]
Same issue.

**Comment by harblaith7** [comment]
Lol are they gonna fix this? Product is unusable otherwise

**Comment by ershisan99** [comment]
Same issue for us, this didn't help: https://github.com/PostHog/posthog/issues/25125

**Comment by tooonuch** [comment]
I found a "workaround", which is a pain to implement especially if you have a lot of events, but it sort of works.
1. Go to "activity" ("explore" tab)
2. find an event you need in the table
3. click on three dots
4. click on "create action from event"
5. In the insights go to actions instead of events, use the one you just created

**Comment by pedromcunha** [comment]
Happening too for us, the workaround mentioned above works but is pretty tedious.

**Comment by tanmoysrt** [comment]
Thanks @ershisan99 , the solution in https://github.com/PostHog/posthog/issues/25125 has worked for me.

In my `docker-compose.yml` file,, I have added this part
```
property-defs-rs:
image: ghcr.io/posthog/posthog/property-defs-rs:master
restart: on-failure
environment:
    DATABASE_URL: 'postgres://posthog:posthog@db:5432/posthog'
    KAFKA_HOSTS: 'kafka:9092'
    SKIP_WRITES: 'false'
    SKIP_READS: 'false'
    FILTER_MODE: 'opt_out'
    CAPTURE_MODE: 'events'

depends_on:
    - db
    - kafka

```

Then, restart all the containers.
It hadn't added events from old data. 
But, once any event are coming in activity, it's getting added in events list.

**Comment by ershisan99** [comment]
> Thanks @ershisan99 , the solution in #25125 has worked for me.
> 
> In my `docker-compose.yml` file, I have added this part
> 
> ```
> property-defs-rs:
>     image: ghcr.io/posthog/posthog/property-defs-rs:master
>     restart: on-failure
>     environment:
>         DATABASE_URL: 'postgres://posthog:posthog@db:5432/posthog'
>         KAFKA_HOSTS: 'kafka:9092'
>         SKIP_WRITES: 'false'
>         SKIP_READS: 'false'
>         FILTER_MODE: 'opt_out'
>         CAPTURE_MODE: 'events'
> 
>     depends_on:
>         - db
>         - kafka
> ```
> 
> Then, restart all the containers. It hadn't added events from old data. But, once any event are coming in activity, it's getting added in events list.

Damn, I'm really glad it worked for you, but I'm really pissed it hasn't for me. They're just not getting added no matter what, even though new events are coming through...

**Comment by tanmoysrt** [comment]
@ershisan99 
It wasn't working for me either.
- I have upgraded posthog to latest version using that upgrade script https://posthog.com/docs/self-host#upgrading
- In the issue suggestion, that `depends_on` part was missing, without that `property-defs-rs` container's process just failed because kafka isn't ready and kafka container takes significant amount of time to start working. After adding that `depends_on` parameter and restarting the containers, it started working.

After these two changes, it started working.

**Comment by ershisan99** [comment]
> @ershisan99 It wasn't working for me either.
> 
> * I have upgraded posthog to latest version using that upgrade script https://posthog.com/docs/self-host#upgrading
> * In the issue suggestion, that `depends_on` part was missing, without that `property-defs-rs` container's process just failed because kafka isn't ready and kafka container takes significant amount of time to start working. After adding that `depends_on` parameter and restarting the containers, it started working.
> 
> After these two changes, it started working.

You sir are a gentleman and a scholar!
I was missing the depends_on part, it's working just fine now.

Thanks!

**Comment by tooonuch** [comment]
@tanmoysrt sorry to bug you, by restart - you mean `docker compose up -d`, or `docker restart $(docker ps -q)`? I assume compose, right?

**Comment by tanmoysrt** [comment]
@tooonuch 
I have just restarted all the containers by `docker compose restart`.
But after adding that part in compose, `docker compose up -d` also should work.

**Comment by ershisan99** [comment]
I wouldn't consider the issue resolve though, the docker-compose file is essentially broken and does not work as it's supposed to

**Comment by tooonuch** [comment]
yeah, this is definitely still a bug, I guess if you upgraded the docker file would get overwritten and the issue would return, as would [this one](https://github.com/PostHog/posthog/issues/26175) which is also fixable with a docker file mod

**Comment by sync-adm** [comment]
It's happening in their cloud in the same way. Data Management and All SelectPickers do not show any type of event despite Activity having the events.

![Image](https://github.com/user-attachments/assets/9832d068-9dfd-4df1-a5cd-fabaa72a816a)

![Image](https://github.com/user-attachments/assets/04e2dc68-1d3b-4dc1-9b2d-afe1a491259b)

**Comment by damjankuznar** [comment]
> It's happening in their cloud in the same way. Data Management and All SelectPickers do not show any type of event despite Activity having the events.

Confirming that I'm still seeing the same issue in our cloud account.

**Comment by rtkachenkotenexti** [comment]
Still experiencing  on b5f6d57, https://github.com/PostHog/posthog/issues/25125 helped to add event properties but no person props

![Image](https://github.com/user-attachments/assets/66a3e82a-cc46-435d-a027-7b1877421972)

**Comment by pauldambra** [comment]
@damjankuznar @sync-adm  for sure this won't be the same issue in cloud as it is in this self hosting report
if you're not seeing events or properties in PostHog cloud please open a support ticket as you absolutely should

**Comment by pauldambra** [comment]
Hey @vamayufei this is specifically about the self-hosted version
if you're experiencing this on cloud you need to visit the in-app support panel

https://us.posthog.com/#panel=support%3Asupport%3A%3A%3Afalse

there can be a delay between ingestion of events and their being available in the UI but it is normally seconds, if that's not true for you then there's probably a setup issue and we can help you out in support

**Comment by pauldambra** [comment]
I'm going to lock this issue because the previous few comments have all been about cloud and definitely aren't the same issue...
